src/storage/src/storage/upload_object/unbuffered.rs

```diff
@@ -32,7 +34,24 @@ where
     /// println!("response details={response:?}");
     /// # Ok(()) }
     /// ```
-    pub async fn send_unbuffered(self) -> crate::Result<Object> {
+    pub async fn send_unbuffered(self) -> Result<Object> {
+        // TODO(#2056) - make idempotency configurable.
+        // Single shot uploads are idempotent only if they have pre-conditions.
+        let idempotent =
+            self.spec.if_generation_match.is_some() || self.spec.if_metageneration_match.is_some();
+        gax::retry_loop_internal::retry_loop(
+            // TODO(#2044) - we need to apply any timeouts here.
```

Comment:
Q: doesn't the retry loop handle this?

Reply:
The loop provides the timeout value, but the value needs to be used.

Reply:
And attempt/overall timeouts come from the retry policy, and are used in the loop, no?

Is this about setting timeout on the reqwest?

Reply:
> And attempt/overall timeouts come from the retry policy,

No.  The attempt timeout should be a combination of the retry policy and any per-request and client configurations. The configurations do not even exist at the moment.

> and are used in the loop, no?

The function called by the retry loop is supposed to apply this combination of retry policy and per-attempt timeouts. We currently are not.

> Is this about setting timeout on the reqwest?

Yes. As the comment says " **apply** any timeouts"